core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -295,17 +297,44 @@ public static void writeShortDecimal(BlockBuilder blockBuilder, long value)
     public static long rescale(long value, int fromScale, int toScale)
     {
         if (toScale < fromScale) {
-            throw new IllegalArgumentException("target scale must be larger than source scale");
+            // Use MAX_SHORT_PRECISION to fit all precision long values
```

Comment:
This could silently round decimals while it failed previously (changes SPI). Please add a new set of methods that do rounding explicitly.

Reply:
I think this should only be possible if there is no rounding, which is tricky since it depends on the actual value having zeros at the end.

Reply:
Okay, will separate them out into new methods. @skrzypo987, can you explain a bit more about your comment. I did not understand it.